frontend/src/router.tsx

```diff
@@ -131,4 +126,33 @@ export const Router = () => {
   // return <RouterProvider router={ROUTER} />;
 };
 
+type RequireAuthProps = {
+  user: ReturnType<typeof useUser>["data"];
+  error: ReturnType<typeof useUser>["error"];
+};
+
+const RequireAuth = ({ user, error }: RequireAuthProps) => {
+  const location = useLocation();
```

Comment:
You could move the useUser hook inside this component, and avoid unnecessary props / pipethrough.